bindings/rust/s2n-tls/src/callbacks.rs

```diff
@@ -142,6 +143,23 @@ impl<F: Future<Output = Result<Config, Error>>> ConfigResolver<F> {
     }
 }
 
+// Useful for propagating [`error::Error`] from the ClientHelloCallback
+// to the Application
+struct ErrorFuture {
+    error: Option<Error>,
+}
+
+impl ConnectionFuture for ErrorFuture {
+    fn poll(
+        mut self: Pin<&mut Self>,
+        _connection: &mut Connection,
+        _ctx: &mut core::task::Context,
+    ) -> Poll<Result<(), Error>> {
+        let err = self.error.take().unwrap();
```

Comment:
error can't be `None`?

Reply:
This is currently only used internally (not pub) to create a future that returns an error immediately so it currently cant be none and we dont ever expect it to be none.

However, in isolation this usage is confusing so i can add a debug assert/expect or `new` here.

Reply:
Once futures return `Poll::Ready` it's generally not recommended to call `poll` again. I've seen quite a few panic when you do that.

Reply:
For example:

https://github.com/tokio-rs/tokio/blob/b42f21ec3e212ace25331d0c13889a45769e6006/tokio/src/future/maybe_done.rs#L70

Reply:
switched to expect with a comment https://github.com/aws/s2n-tls/pull/3687/commits/21e71fcd4d487ff3de707cb686fd34b58203b797